packages/gensx-core/src/utils/project-config.ts

```diff
@@ -0,0 +1,51 @@
+import { readFileSync } from "node:fs";
+import path from "node:path";
+
+import { z } from "zod";
+
+// Define schema for gensx.yaml
+const ProjectConfigSchema = z.object({
+  projectName: z.string(),
+  environmentName: z.string().optional(),
+  description: z.string().optional(),
+});
+
+export type ProjectConfig = z.infer<typeof ProjectConfigSchema>;
+
+/**
+ * Get the path to the gensx.yaml file
+ */
+export function getProjectConfigPath(dir: string): string {
+  return path.join(dir, "gensx.yaml");
+}
```

Comment:
I wonder if the logic for finding a project file is robust enough. I think this should be fine as all of our examples have the project file at the same level as the package.json where a user will be running commands (the directory we'll detect as entrypoint). But wouldn't be surprised if we discover some edge cases that require recursively searching parent directories for this file. Probably fine for now.

Reply:
Agree, this will definitely bite someone.

Reply:
I added an issue to look into this: https://github.com/gensx-inc/gensx/issues/648